bin/install-pre-push-hooks.sh

```
@@ -0,0 +1,49 @@
+
+# Install for root repository
+ROOT_HOOK_DIR=$(git -C "$ROOT_DIR" rev-parse --git-path hooks)
+install_hook "$ROOT_HOOK_DIR"
+echo "Installed pre-push hook for root repo"
+ROOT_EXCLUDE=$(git -C "$ROOT_DIR" rev-parse --git-path info/exclude)
+ensure_nx_ignore "$ROOT_EXCLUDE"
+echo "Ensured .nx ignore for root repo"
+
+# Install for each submodule recursively
+if git -C "$ROOT_DIR" rev-parse --git-dir >/dev/null 2>&1; then
+  HOOK_SOURCE="$HOOK_SOURCE" git -C "$ROOT_DIR" submodule foreach --quiet --recursive "
```

Shellcheck (0.11.0)

[warning] This assignment is only seen by the forked process.

(SC2097)